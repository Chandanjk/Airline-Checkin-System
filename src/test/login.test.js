import React from "react";
import * as renderer from "react-test-renderer";
import Login from "../components/login/login";
import { waitFor } from "@testing-library/react";

it("Component is rendering without crashing", async () => {
  let tree;
  renderer.act(() => {
    tree = renderer.create(
      <Login
        handleLogin={jest.fn()}
        onChange={jest.fn()}
        successGoogleLogin={jest.fn()}
        failureGoogleLogin={jest.fn()}
      />
    );
  });
  await waitFor(() => {
    expect(tree).toMatchSnapshot();
  });
});
