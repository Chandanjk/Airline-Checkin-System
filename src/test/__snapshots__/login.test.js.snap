// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is rendering without crashing 1`] = `
<div
  className="wrapper fadeInDown"
>
  <span>
    <h1
      className="heading1 active"
    >
       AIRLINE CHECK IN SYSTEM 
    </h1>
  </span>
  <div
    id="formContent"
  >
    <h2
      className="heading2 active"
    >
       Sign In 
    </h2>
    <form>
      <input
        className="inputFields fadeIn second"
        id="login"
        name="email"
        onChange={[MockFunction]}
        placeholder="login"
        required={true}
        type="text"
      />
      <input
        className="inputFields fadeIn third"
        id="password"
        name="password"
        onChange={[MockFunction]}
        placeholder="password"
        required={true}
        type="password"
      />
      <input
        className="inputBtn fadeIn fourth"
        onClick={[MockFunction]}
        type="button"
        value="Log In"
      />
      <div
        className="fadeIn fourth"
        id="googleLogin"
        style={
          {
            "padding": "0",
          }
        }
      >
        <div
          style={
            {
              "height": 40,
            }
          }
        />
      </div>
    </form>
  </div>
</div>
`;
